pkg/test/framework/components/echo/echotest/setup.go

```diff
@@ -41,21 +48,24 @@ func (t *T) setup(ctx framework.TestContext, srcInstances echo.Instances) {
 	}
 }
 
-// SetupForPair runs the given function in the source + destination deployment context. For example, given apps a, b,
-// and c in 2 clusters, these tests would all run before the the context is cleaned up:
+// SetupForPair runs the given function in the source + destination deployment context. The setup function
+// takes a []echo.Instances, where each element contains all the instances gropued by destination deployment.
+// When using Run there will always be 1 destination deployment. When using RunForN, the length will always be N.
+//
+// Example of how long this setup lasts before the given context is cleaned up:
 //     a/to_b/from_cluster-1
 //     a/to_b/from_cluster-2
 //     cleanup...
 //     a/to_b/from_cluster-2
 //     ...
-func (t *T) SetupForPair(setupFn func(ctx framework.TestContext, src echo.Instances, dst echo.Instances) error) *T {
+func (t *T) SetupForPair(setupFn pairSetupFn) *T {
```

Comment:
An alternative would be leaving this as, and the "pair" would have a flattened list of multiple deployments.
If the setup cares about the deployments, they can un-flatten by calling the `Deployments` method we already have.